app/controllers/api/providers_controller.rb

```diff
@@ -85,6 +85,11 @@ def import_vm_resource(type, id = nil, data = {})
       end
     end
 
+    def options
+      ems_class = ExtManagementSystem.descendants.find { |ems| ems.ems_type == params[:ems_type] }
+      render_options(:providers, ems_class.respond_to?(:provider_settings) ? ems_class.provider_settings : {})
```

Comment:
It looks like you don't render anything when you don't provide the `ems_type` param. Did you you consider rendering all the EMSs and their provider settings and leaving it to the client to figure out?